File: app/src/main/java/at/shockbytes/coins/network/conversion/CurrencyConversionApi.java
package at.shockbytes.coins.network.conversion;

import at.shockbytes.coins.currency.CurrencyConversionRates;
import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * @author Martin Macheiner
 *         Date: 01.07.2017.
 */

public interface CurrencyConversionApi {

    String SERVICE_ENDPOINT = "http://api.fixer.io/";

    @GET("latest")
    Observable<CurrencyConversionRates> getCurrencyConversionRates(@Query("base") String from,
                                                                   @Query("symbols") String to);


}

File: app/src/main/java/at/shockbytes/coins/network/conversion/CurrencyConversionApi.kt
package at.shockbytes.coins.network.conversion

import at.shockbytes.coins.currency.conversion.CurrencyConversionRates
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author Martin Macheiner
 * Date: 01.07.2017.
 */

interface CurrencyConversionApi {

    @GET("latest")
    fun getCurrencyConversionRates(@Query("base") from: String,
                                   @Query("symbols") to: String): Observable<CurrencyConversionRates>

    companion object {

        val SERVICE_ENDPOINT = "http://api.fixer.io/"
    }


}
